python/cudf/cudf/core/buffer.py

```diff
@@ -203,7 +204,7 @@ def confirm_1d_contiguous(array_interface):
     shape = array_interface["shape"]
     itemsize = cudf.dtype(array_interface["typestr"]).itemsize
     typestr = array_interface["typestr"]
-    if typestr not in ("|i1", "|u1"):
+    if typestr not in ("|i1", "|u1", "|b1"):
```

Comment:
Wonder if we should just loop through CuPy arrays of these types (and one non-conforming one)

Reply:
Is there a publicly exposed cupy array types we can loop through?

Reply:
Would look at [`numpy.typecodes`]( https://github.com/numpy/numpy/blob/abd47fc824ca9cdd2c03063b5d294039a2f6f0d1/numpy/core/numerictypes.py#L531-L539 ). Each single character maps to a type. So can be passed to `numpy.dtype(...)` or as a `dtype` kwarg.

Reply:
I think we need to only check here for types that are 8-bit types, so we won't need to loop through rest of the all cuda array interface supported types. Because Buffer's can only be constructed with 8-bit types.